dataclasses library breaks docker build

To reproduce add `RUN python3.8 -m hmalib.lambdas.pdq.pdq_hasher` to bottom of Dockerfile and run `make docker`

stack trace:
```
 > [prod 4/4] RUN python3.8 -m hmalib.lambdas.pdq.pdq_hasher:
#11 1.001 /hma-deps/pdqhash/__init__.py:3: UserWarning: Hash vector order changed between version 0.1.8 and 0.2.0. See https://github.com/faustomorales/pdqhash-python/issues/1 for more details.
#11 1.001   warnings.warn(
#11 1.001 Traceback (most recent call last):
#11 1.001   File "/var/lang/lib/python3.8/runpy.py", line 194, in _run_module_as_main
#11 1.001     return _run_code(code, main_globals, None,
#11 1.001   File "/var/lang/lib/python3.8/runpy.py", line 87, in _run_code
#11 1.001     exec(code, run_globals)
#11 1.001   File "/var/task/hmalib/lambdas/pdq/pdq_hasher.py", line 15, in <module>
#11 1.001     from hmalib.models import PipelinePDQHashRecord
#11 1.001   File "/var/task/hmalib/models.py", line 24, in <module>
#11 1.001     class PDQRecordBase(DynamoDBItem):
#11 1.001   File "/hma-deps/dataclasses.py", line 958, in dataclass
#11 1.001     return wrap(_cls)
#11 1.001   File "/hma-deps/dataclasses.py", line 950, in wrap
#11 1.001     return _process_class(cls, init, repr, eq, order, unsafe_hash, frozen)
#11 1.001   File "/hma-deps/dataclasses.py", line 800, in _process_class
#11 1.001     cls_fields = [_get_field(cls, name, type)
#11 1.001   File "/hma-deps/dataclasses.py", line 800, in <listcomp>
#11 1.001     cls_fields = [_get_field(cls, name, type)
#11 1.001   File "/hma-deps/dataclasses.py", line 659, in _get_field
#11 1.001     if (_is_classvar(a_type, typing)
#11 1.001   File "/hma-deps/dataclasses.py", line 550, in _is_classvar
#11 1.001     return type(a_type) is typing._ClassVar
#11 1.001 AttributeError: module 'typing' has no attribute '_ClassVar'
------
executor failed running [/bin/sh -c python3.8 -m hmalib.lambdas.pdq.pdq_hasher]: exit code: 1
make: *** [docker] Error 1
```

--- a/hasher-matcher-actioner/Dockerfile
+++ b/hasher-matcher-actioner/Dockerfile
@@ -1,16 +1,7 @@
 # Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
 
-ARG DEPS_PATH=/hma-deps
-ARG CODE_DIRECTORY=/hma-lambda
-
 FROM public.ecr.aws/lambda/python:3.8 as build
 RUN yum install -y gcc gcc-c++
-COPY requirements.txt .
-ARG DEPS_PATH
-RUN pip install --no-cache-dir -r requirements.txt --target "${DEPS_PATH}"
-
-FROM public.ecr.aws/lambda/python:3.8 as prod
-ARG DEPS_PATH
-COPY --from=build $DEPS_PATH $DEPS_PATH
-ENV PYTHONPATH $DEPS_PATH
+COPY setup.py ./setup.py
 COPY hmalib ./hmalib
+RUN pip install --no-cache-dir .

--- a/hasher-matcher-actioner/hmalib/lambdas/fetcher.py
+++ b/hasher-matcher-actioner/hmalib/lambdas/fetcher.py
@@ -39,7 +39,6 @@ dynamodb = boto3.resource("dynamodb")
 s3 = boto3.resource("s3")
 
 
-
 @dataclass
 class FetcherConfig:
     """

--- a/hasher-matcher-actioner/requirements.txt
+++ b/hasher-matcher-actioner/requirements.txt
@@ -1,5 +1,1 @@
-boto3
-boto3-stubs[essential]==1.17.14.0
-threatexchange[faiss,pdq_hasher]>=0.0.15
-bottle
-apig_wsgi
+.

--- a/hasher-matcher-actioner/setup.py
+++ b/hasher-matcher-actioner/setup.py
@@ -1,0 +1,18 @@
+#!/usr/bin/env python
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+
+from setuptools import setup
+
+
+setup(
+    name="hmalib",
+    description="Python Library for Facebook ThreatExchange",
+
+    install_requires=[
+        "boto3",
+        "boto3-stubs[essential]==1.17.14.0",
+        "threatexchange[faiss,pdq_hasher]>=0.0.15",
+        "bottle",
+        "apig_wsgi",
+    ],
+)

--- a/hasher-matcher-actioner/setup.py
+++ b/hasher-matcher-actioner/setup.py
@@ -6,7 +6,7 @@ from setuptools import setup
 
 setup(
     name="hmalib",
-    description="Python Library for Facebook ThreatExchange",
+    description="Convenience package for hmalib. Probably don't distribute it.",
 
     install_requires=[
         "boto3",

--- a/hasher-matcher-actioner/setup.py
+++ b/hasher-matcher-actioner/setup.py
@@ -7,7 +7,6 @@ from setuptools import setup
 setup(
     name="hmalib",
     description="Convenience package for hmalib. Probably don't distribute it.",
-
     install_requires=[
         "boto3",
         "boto3-stubs[essential]==1.17.14.0",

--- a/python-threatexchange/setup.py
+++ b/python-threatexchange/setup.py
@@ -57,6 +57,7 @@ setup(
     install_requires=[
         "python-Levenshtein",
         "requests",
+        "dataclasses",
     ],
     extras_require=extras_require,
     entry_points={"console_scripts": ["threatexchange = threatexchange.cli.main:main"]},

